yt/data_objects/data_containers.py

```diff
@@ -1428,52 +1427,72 @@ def _field_type_state(self, ftype, finfo, obj=None):
         explicit_fields = []
         for field in fields:
             if field in self._container_fields:
                 explicit_fields.append(field)
                 continue
-            if isinstance(field, tuple):
-                if (
-                    len(field) != 2
-                    or not isinstance(field[0], str)
-                    or not isinstance(field[1], str)
-                ):
-                    raise YTFieldNotParseable(field)
-                ftype, fname = field
-                finfo = self.ds._get_field_info(ftype, fname)
-            elif isinstance(field, DerivedField):
-                ftype, fname = field.name
-                finfo = field
-            else:
-                fname = field
-                finfo = self.ds._get_field_info("unknown", fname)
-                if finfo.sampling_type == "particle":
-                    ftype = self._current_particle_type
-                    if hasattr(self.ds, "_sph_ptypes"):
-                        ptypes = self.ds._sph_ptypes
-                        if finfo.name[0] in ptypes:
-                            ftype = finfo.name[0]
-                        elif finfo.alias_field and finfo.alias_name[0] in ptypes:
-                            ftype = self._current_fluid_type
-                else:
-                    ftype = self._current_fluid_type
-                    if (ftype, fname) not in self.ds.field_info:
-                        ftype = self.ds._last_freq[0]
-
-                # really ugly check to ensure that this field really does exist somewhere,
-                # in some naming convention, before returning it as a possible field type
-                if (
-                    (ftype, fname) not in self.ds.field_info
-                    and (ftype, fname) not in self.ds.field_list
-                    and fname not in self.ds.field_list
-                    and (ftype, fname) not in self.ds.derived_field_list
-                    and fname not in self.ds.derived_field_list
-                    and (ftype, fname) not in self._container_fields
-                ):
-                    raise YTFieldNotFound((ftype, fname), self.ds)
+
+            ftype, fname = self._tupleize_field(field)
+            # print(field, " : ",ftype, fname)
+            finfo = self.ds._get_field_info(ftype, fname)
+
+            # really ugly check to ensure that this field really does exist somewhere,
+            # in some naming convention, before returning it as a possible field type
+            if (
+                (ftype, fname) not in self.ds.field_info
+                and (ftype, fname) not in self.ds.field_list
+                and fname not in self.ds.field_list
+                and (ftype, fname) not in self.ds.derived_field_list
+                and fname not in self.ds.derived_field_list
+                and (ftype, fname) not in self._container_fields
+            ):
+                raise YTFieldNotFound((ftype, fname), self.ds)
```

Comment:
So here I didn't really try to simplify the logic but it seems weird to me that this check is even a thing. I think that this line
```python
finfo = self.ds._get_field_info(ftype, fname)
```
would fail already if the field isn't present. @matthewturk, I think you authored this check, do you believe it is still relevant after the refactor ?